lib/prism/translation/parser/lexer.rb

```diff
@@ -595,9 +596,9 @@ def calculate_heredoc_whitespace(heredoc_token_index)
           previous_line = -1
           result = Float::MAX
 
-          while (lexed[next_token_index] && next_token = lexed[next_token_index][0])
+          while (next_token, _ = lexed[next_token_index])
```

Comment:
I don't think the behaviour of these two constructs (old and new) will be the same in every instance. For example:
```ruby
irb> foo = [nil, 1]
#=> [nil, 1]

irb> while (i, _ = foo); puts "42"; break; end
42
#=> nil

irb> while (i = foo&.first); puts "42"; break; end
#=> nil
```

The same would be true for any assignments in `if` conditions as well.

I would very much like to see `&.first` instead of destructuring used to safely get these values, since its behaviour would be the same as `[0]` (but safe against `nil` values) and I also find that it is more readable.

Reply:
Good catch, I was expecting the destructure to be falsy

Reply:
The token being `nil` would be a bug in prism I believe (and probably blow up in a different place later). I can see now how this isn't quite the same but practially the difference should be zero.

Anyways, I changed it.

Reply:
I was just saying that this difference shouldn't matter for the data that prism provides. It's always [token, some_internal_state], if there's no token (`nil`, then that should not have been part of `lexed` in the first place.